tests/api/test_web_routes_task_shapes.py

```
@@ -351,3 +351,441 @@ async def test_execute_task_from_project_preserves_idle_no_ready_work_truth():
+    core = SimpleNamespace(
+        project_manager=SimpleNamespace(
+            create_project_async=AsyncMock(return_value=project),
+            sync_blueprint=Mock(),
+            get_ready_tasks_async=AsyncMock(return_value=[]),
+            list_tasks=Mock(return_value=[]),
+            storage=storage,
+        )
+    )
+
+    with patch("penguin.web.services.projects.BlueprintParser", return_value=parser):
+        with pytest.raises(HTTPException) as exc_info:
```

Ruff (0.15.11)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)